x-pack/plugins/security/public/nav_control/nav_control_component.tsx

```diff
@@ -32,174 +35,127 @@ export interface UserMenuLink {
-        {buttonContents}
-      </EuiHeaderSectionItemButton>
-    );
-
-    const isAnonymousUser = authenticatedUser?.authentication_provider.type === 'anonymous';
-    const items: EuiContextMenuPanelItemDescriptor[] = [];
-
-    if (userMenuLinks.length) {
-      const userMenuLinkMenuItems = userMenuLinks
-        .sort(({ order: orderA = Infinity }, { order: orderB = Infinity }) => orderA - orderB)
-        .map(({ label, iconType, href }: UserMenuLink) => ({
-          name: label,
-          icon: <EuiIcon type={iconType} size="m" />,
-          href,
-          'data-test-subj': `userMenuLink__${label}`,
-        }));
-      items.push(...userMenuLinkMenuItems);
-    }
-
-    if (!isAnonymousUser) {
-      const hasCustomProfileLinks = userMenuLinks.some(({ setAsProfile }) => setAsProfile === true);
-      const profileMenuItem = {
-        name: (
-          <FormattedMessage
-            id="xpack.security.navControlComponent.editProfileLinkText"
-            defaultMessage="{profileOverridden, select, true{Preferences} other{Profile}}"
-            values={{ profileOverridden: hasCustomProfileLinks }}
-          />
-        ),
-        icon: <EuiIcon type={hasCustomProfileLinks ? 'controlsHorizontal' : 'user'} size="m" />,
-        href: editProfileUrl,
-        'data-test-subj': 'profileLink',
-      };
-
-      // Set this as the first link if there is no user-defined profile link
-      if (!hasCustomProfileLinks) {
-        items.unshift(profileMenuItem);
-      } else {
-        items.push(profileMenuItem);
-      }
-    }
-
-    const logoutMenuItem = {
-      name: isAnonymousUser ? (
-        <FormattedMessage
-          id="xpack.security.navControlComponent.loginLinkText"
-          defaultMessage="Log in"
+export const SecurityNavControl: FunctionComponent<SecurityNavControlProps> = ({
+  editProfileUrl,
+  logoutUrl,
+  userMenuLinks$,
+}) => {
+  const userMenuLinks = useObservable(userMenuLinks$) ?? [];
+  const [isOpen, setIsOpen] = useState(false);
+
+  const userProfile = useUserProfile();
+
+  const displayName = userProfile.value ? getUserDisplayName(userProfile.value.user) : '';
+
+  const button = (
+    <EuiHeaderSectionItemButton
+      aria-controls="headerUserMenu"
+      aria-expanded={isOpen}
+      aria-haspopup="true"
+      aria-label={i18n.translate('xpack.security.navControlComponent.accountMenuAriaLabel', {
+        defaultMessage: 'Account menu',
+      })}
+      onClick={() => setIsOpen((value) => (userProfile.value ? !value : false))}
+      data-test-subj="userMenuButton"
+    >
+      {userProfile.value ? (
+        <UserAvatar
+          user={userProfile.value.user}
+          avatar={userProfile.value.data.avatar}
+          size="s"
+          data-test-subj="userMenuAvatar"
         />
       ) : (
+        <EuiLoadingSpinner size="m" />
+      )}
+    </EuiHeaderSectionItemButton>
+  );
+
+  const isAnonymousUser = false; // authenticatedUser?.authentication_provider.type === 'anonymous';
```

Comment:
TODO: Make separate call to `/internal/security/me` or return authentication_provider as part of user profile API.